module/zfs/arc.c

```diff
@@ -1331,13 +1391,30 @@ arc_buf_lsize(arc_buf_t *buf)
 	return (HDR_GET_LSIZE(buf->b_hdr));
 }
 
+boolean_t
+arc_is_encrypted(arc_buf_t *buf)
+{
+	return (ARC_BUF_ENCRYPTED(buf) != 0);
+}
+
 enum zio_compress
 arc_get_compression(arc_buf_t *buf)
 {
 	return (ARC_BUF_COMPRESSED(buf) ?
 	    HDR_GET_COMPRESS(buf->b_hdr) : ZIO_COMPRESS_OFF);
 }
 
+/*
+ * Returns the compression that should be used for compressed buffers. If
+ * ARC compression is disabled, returns ZIO_CRYPT_OFF.
```

Comment:
Will fix. This function replaced `HDR_GET_COMPRESS()` in most cases, so I should probably add something there mentioning that most consumers will want this function instead.